File: brute_dns.py
#!/usr/bin/env python3

import sys

import dns.resolver


resolver = dns.resolver.Resolver()

try:
	(target, wordlist) = (sys.argv[1], sys.argv[2])
except:
	print(f'Usage: python3 {sys.argv[0]} <domain> <wordlist>')
	sys.exit(1)

try:
	with open(wordlist, 'r') as file:
		subdomains = file.read().splitlines()
except Exception as e:
	print(f'Error while opening the file: {e}')
	sys.exit(1)

for subdomain in subdomains:
	try:
		sub_target = f'{subdomain}.{target}'
		results = resolver.resolve(sub_target, 'A')
		for result in results:
			print(f'{result}\t{sub_target}')
	except:
		pass

File: brute_dns.pl
#!/usr/bin/env perl
# Usage: perl ./brute_dns.pl <domain> <wordlist>
# Makes a brute-forcing of subdomains based in a wordlist and returns the respectives IP addresses.

use 5.32.0;
use Net::DNS;

my $resolver = Net::DNS::Resolver->new();
my ($target, $wordlist) = @ARGV;

die "Usage: $0 <domain> <wordlist>\n" 
unless defined $target and defined $wordlist;

open my $fh, '<', $wordlist or die "Couldn't open the file $wordlist: $!\n";
chomp(my @subdomains = <$fh>);
close $fh;

foreach (@subdomains) {
  my $sub_target = "$_.$target";
  my $query = $resolver->search($sub_target, 'A');
  if ($query) {
    foreach my $rr ($query->answer) {
      next unless $rr->type eq 'A';
      say $rr->address . "\t$sub_target";
    }
  }
}
